### DISCUSSION TITLE
What is the difference between text-body-secondary and text-secondary?

### DISCUSSION THREAD

**Discussion by carlosmintfan** [twbs/bootstrap#40650] [Q&A] [answered]
What's the difference between `text-body-secondary` and `text-secondary` (without body)?
They have a different opacity or so...but why?

**Comment by julien-deramond** [comment, marked as answer]
I've created a [CodePen](https://codepen.io/julien-deramond/pen/poXyxzB) to visualize the differences and serve as a reference.

To display normal text, such as paragraphs, articles, long sections, etc. you’ll typically use `.text-body`. If you need lighter text, you can use `.text-body-secondary`, and for even lighter text, `.text-body-tertiary`. Even if you customize Bootstrap, it's likely that `.text-body-*` will remain shades of black in light mode.

However, the concepts of primary and secondary colors are different. These terms are used to define your brand’s primary and secondary colors. In Bootstrap, primary is blue, and secondary is black. In my example, if you want to create a tag, you’ll have a primary tag with a subtle primary background, primary border, and primary text. The same applies to your secondary color, which corresponds to the `text-secondary` color.

This can be confusing for users who do not customize Bootstrap, as the secondary color is gray and often used for regular text. However, for a different brand, the secondary color could be yellow. In that case, your `text-body-*` classes would be black and gray (in light mode), while `text-secondary` would be yellow.

The semantics are important.

Additionally, in our current implementation, `text-primary` and `text-secondary` do not change colors between light and dark modes, so the accessibility color contrast will differ from `text-body-*` on a `bg-body` background.

**Reply by carlosmintfan** [reply]
Thank you for your detailed explanation! I still have some questions:
1. Is text without classes the same as .text-body?
2. And then there's this bg-light class (yes, bg, not text). It seems to me that it's the same as bg-body-secondary (if not tertiary, I don't know right now); however it looks stupid in dark mode. So is it recommended to always use bg-body-secondary/tertiary instead of bg-light? Why do we even have both options - I think the body classes were added later, right?

**Reply by julien-deramond** [reply]
> 1. Is text without classes the same as .text-body?

By default, we define `<body>` to have `color: var(--bs-body-color)` which is the same color used by `.text-body`. So, generally speaking, text without additional classes will have the same color as `.text-body`, adapting to both light and dark modes. However, depending on your interface's complexity, you might need to apply these color rules within a specific sub-div, making the `.text-body` class useful in those scenarios.

> 2. And then there's this bg-light class [...] it looks stupid in dark mode

Classes like `.bg-light` and `.bg-dark` behave similarly to the primary, secondary, and success concepts. This means their colors remain constant regardless of light or dark mode. In contrast, `.bg-body*` classes adjust their colors to suit both light and dark modes.

> So is it recommended to always use bg-body-secondary/tertiary instead of bg-light? Why do we even have both options

The choice depends on your use cases across different color modes. If you don't need to accommodate color mode changes, you can use basic classes like `.bg-light`. However, if your interface needs to adapt to light and dark modes, using classes like `.bg-body`, `.bg-body-secondary` or `.bg-body-tertiary` is more appropriate.
